actions/cwrhelpers.py

```diff
@@ -10,20 +10,59 @@
 
 from charmhelpers.core import hookenv  # noqa: E402
 from jenkins import NotFoundException  # noqa: E402
-from theblues import charmstore  # noqa: E402
+from theblues.charmstore import CharmStore  # noqa: E402
+from theblues.errors import EntityNotFound, ServerError  # noqa: E402
 
 
-def app_from_bundle(bundle, charm):
+def app_from_bundle(bundle_name, charm_name):
     '''Return the app name used in the given bundle for a given charm.'''
-    cs = charmstore.CharmStore()
-    bundle_yaml = cs.files(bundle, filename='bundle.yaml', read_file=True)
+    cs = CharmStore()
+    bundle_yaml = cs.files(bundle_name, filename='bundle.yaml', read_file=True)
     yaml_contents = yaml.safe_load(bundle_yaml)
     for app, app_config in yaml_contents['services'].items():
-        if charm in app_config['charm']:
+        if charm_name in app_config['charm']:
             return app
     return None
 
 
+def fetch_reference_bundle(charm_name):
+    try:
+        tests_yaml = CharmStore().files(charm_name,
+                                        filename='tests/tests.yaml',
+                                        read_file=True)
+        tests_yaml = yaml.safe_load(tests_yaml)
+        return tests_yaml.get('reference-bundle')
+    except EntityNotFound:
+        return None  # probably doesn't have a test.yaml
+    except (ServerError, yaml.YAMLError) as e:
+        hookenv.log('Unable to load test.yaml from %s: %s' % (
+            charm_name, e), hookenv.ERROR)
+
+
+def get_reference_bundle():
+    # If we have a reference bundle, determine the app name used for our charm
+    charm_name = hookenv.action_get("charm-name")
+    if hookenv.action_get("reference-bundle"):
+        bundle_name = hookenv.action_get("reference-bundle")
+    else:
+        # try to get the reference bundle from tests.yaml
+        bundle_name = fetch_reference_bundle(charm_name)
+
+    if bundle_name:
+        if bundle_name.startswith('bundle:'):
+            # normalize bundle: style URL (why do we even accept this?)
```

Comment:
I think this was a hold-over from before we had a bundle "namespace" (e.g., `cs:bundle/wiki-simple`), so I'm somewhat inclined to deprecate it in BT and not carry it forward here, since it's non-standard.